cloudevents/sdk/http_events.py

```diff
@@ -64,8 +65,47 @@ def from_http(
         if data_unmarshaller is None:
             data_unmarshaller = _json_or_string
 
-        event = marshaller.NewDefaultHTTPMarshaller().FromRequest(
-            v1.Event(), headers, data, data_unmarshaller
+        headers = {key.lower(): value for key, value in headers.items()}
+
+        content_type = headers.get("content-type", None)
+
+        specversion = None
+        marshall = marshaller.NewDefaultHTTPMarshaller()
+
+        for converter in marshall.http_converters:
+            # marshall atm has two converters, one for binary and another for
```

Comment:
nit: avoid abbreviations in comments, makes it difficult for folks with less skill reading English.

Reply:
I have a habit of doing that, good catch!